server/src/monitoring/monitoring.js

```diff
@@ -1,15 +1,44 @@
-var yml2js = require('js-yaml'),
+var camel = require('camel-case'),
+    xml2js = require('xml2js').parseString,
+    yml2js = require('js-yaml'),
     winston = require('winston'),
     fs = require('fs'),
     NEW_RELIC_CONFIG = '/agents/newrelic/newrelic.yml';
 
-// NEW RELIC
+
+// FIRST, LETS CHECK FOR APPDYNAMICS
+if (process.env.YTENV_USE_APPDYNAMICS) {
+
+    var xmlFile;
+    try {
+        xmlFile = String(fs.readFileSync('/agents/appdynamics-jvm/conf/controller-info.xml'));
+    } catch (err) {
+        winston.error('Could not read appdynamics config XML.', err.message);
+    }
+    xml2js(xmlFile, function (err, result) {
+        if (err) {
+            winston.error('Could not parse appdynamics config XML. Error: %s. Content: %s.', err.message, xmlFile);
+            return;
+        }
+        var config = Object
```

Comment:
i didn't change here anything on that xml -> json reading part. so what do you mean exactly?